### ISSUE TITLE
Hyperparameters optimization

### ISSUE THREAD

**Issue by mariehbourget** [axondeepseg/axondeepseg#304]
In order to retrain the SEM model, I need to identify the hyperparameters to optimize.

I found several in the training notebook (and config_network files) that are no longer in the code base (as of Keras integration I think) :
```
"learning_rate_decay_activate"
"learning_rate_decay_period"
"learning_rate_decay_type"
"learning_rate_decay_rate"
"batch_norm_decay_decay_activate"
"batch_norm_decay_ending_decay"
"batch_norm_decay_decay_period"
"weighted_cost-activate"
"weighted_cost-balanced_activate"
"weighted_cost-balanced_weights"
"weighted_cost-boundaries_sigma"
"weighted_cost-boundaries_activate"
```

We should remove them from the notebooks (and future config_network files).

Also, in the Adam optimizer, the `learning_rate` is now fixed to the value defined in the config_network file. There is no learning rate decay during training (as explain [here](https://github.com/neuropoly/axondeepseg/pull/221#issuecomment-503738167))

So except for the architecture of the Unet and patchsize, the only parameters I can optimized are
```
"downsampling"
"batch_size"
"dropout"
"learning_rate"
```
and try different combinations of data augmentation.

Should I proceed with those or include modifications in the architecture of the network?
If so, we will have different architectures for SEM and TEM, is this a problem?

**Comment by mariehbourget** [comment]
As a reference , the following parameters were used in `train_network.py` (ex: in [test_performance_tf_VS-Keras branch](https://github.com/neuropoly/axondeepseg/blob/test_performace_tf-VS-Keras/AxonDeepSeg/train_network.py)) :
```
"learning_rate_decay_activate"
"learning_rate_decay_period"
"learning_rate_decay_type"
"learning_rate_decay_rate"
"batch_norm_decay_decay_activate"
"batch_norm_decay_ending_decay"
"batch_norm_decay_decay_period"
"weighted_cost-activate"
```
And I think the following were used in `generate_dict_weights` function in `train_network_tools.py` (no longer in the code) [here](https://github.com/neuropoly/axondeepseg/blob/e1d37e0bbe8093045059f2a85a37d92904d74ac2/AxonDeepSeg/train_network_tools.py#L50) :
```
"weighted_cost-balanced_activate"
"weighted_cost-balanced_weights"
"weighted_cost-boundaries_sigma"
"weighted_cost-boundaries_activate"
```

**Comment by mathieuboudreau** [comment]
@mariehbourget I think this can be closed now that we moved the training to IVADOMED?
